src/global/variables/clauses/DoubleLiteral.java

```diff
@@ -0,0 +1,64 @@
+package global.variables.clauses;
+
+import global.tools.InvalidClauseException;
+import global.variables.Clause;
+
+public class DoubleLiteral extends Clause {
+    private String mode;
+    public static final String RANGE = "range";
+    double upper;
+    double lower;
+
+
+    public DoubleLiteral(String name, double lower, double upper) {
```

Comment:
I think `name` is usually the last parameter

Reply:
will change that asap